sql/core/src/test/scala/org/apache/spark/sql/DatasetOptimizationSuite.scala

```diff
@@ -32,9 +32,9 @@ class DatasetOptimizationSuite extends QueryTest with SharedSparkSession {
   test("SPARK-26619: Prune the unused serializers from SerializeFromObject") {
     val data = Seq(("a", 1), ("b", 2), ("c", 3))
     val ds = data.toDS().map(t => (t._1, t._2 + 1)).select("_1")
-    val serializer = ds.queryExecution.optimizedPlan.collect {
+    val serializer = ds.queryExecution.optimizedPlan.collectFirst {
```

Comment:
Personally think the following approach might be a bit better:

```scala
val serializerOpt = ds.queryExecution.optimizedPlan.collectFirst {
      case s: SerializeFromObject => s
    }
    assert(serializerOpt.isDefined)
    assert(serializerOpt.get.serializer.size == 1)
```

Reply:
Follow this suggestion